src/tests/illumidesk/apps/test_jupyterhub_base_config.py

```diff
@@ -0,0 +1,46 @@
+
+import os
+import pytest
+
+from distutils.util import strtobool
+
+from traitlets.config import Config
+
+
+def test_jupyterhub_base_config(setup_jupyterhub_db, setup_jupyterhub_config_base):
+    """
+    Ensure all environment variables for the base config are set.
+    """
+    c = Config()
+    c.JupyterHub.base_url = os.environ.get('JUPYTERHUB_BASE_URL')
+    # JupyterHub postgres settings
+    c.JupyterHub.db_url = 'postgresql://{user}:{password}@{host}:{port}/{db}'.format(
+        user=os.environ.get('POSTGRES_JUPYTERHUB_USER'),
+        password=os.environ.get('POSTGRES_JUPYTERHUB_PASSWORD'),
+        host=os.environ.get('POSTGRES_JUPYTERHUB_HOST'),
+        port=os.environ.get('POSTGRES_JUPYTERHUB_PORT'),
+        db=os.environ.get('POSTGRES_JUPYTERHUB_DB'),
+    )
+    c.JupyterHub.shutdown_on_logout = bool(strtobool(os.environ.get('JUPYTERHUB_SHUTDOWN_ON_LOGOUT')))
+    c.Authenticator.admin_users = {os.environ.get('JUPYTERHUB_ADMIN_USER')}
+    c.Spawner.image = os.environ.get('DOCKER_END_USER_IMAGE')
+    c.Spawner.cpu_limit = float(os.environ.get('SPAWNER_CPU_LIMIT'))
+    c.Spawner.mem_limit = os.environ.get('SPAWNER_MEM_LIMIT')
+    c.DockerSpawner.network_name = os.environ.get('DOCKER_NETWORK_NAME')
+    docker_spawn_command = os.environ.get('DOCKER_SPAWN_CMD')
+    exchange_dir = os.environ.get('EXCHANGE_DIR')
+    notebook_dir = os.environ.get('DOCKER_NOTEBOOK_DIR')
+
+    assert c.Authenticator.admin_users == {'admin0'}
```

Comment:
@jgwerner In my opinion we're testing things that jupyterhub does or that are part of the system. For example this test only confirm that a value is set from an environ variable so in the runtime the env var could be changed.. Maybe it makes more sense in an integration test because it will ensure that our code reads specific values set in a deployment

Reply:
@netoisc that is a good point. The only reason I added this test (again would agree that it's not very useful) is because I made the silly mistake of misspelling the env var after the copy/paste of the hosts file. Then, the jupyterhub started to fail (quite miserably I might add). I'm sure there is a better way to do this but thought we could add this in the meantime.